rules/firestore.test.json

```diff
@@ -154,11 +154,19 @@ service cloud.firestore {
             // A player can perform a move
             let isUserTurn = (part.turn % 2 == 0 && userIsPlayerZero(part)) ||
                 (part.turn % 2 == 1 && userIsPlayerOne(part));
-            return part.listMoves != update.listMoves &&
-                isValidPartUpdateDuringGame(part) &&
-                safeTurnIncrease(part.turn, update.turn) &&
-                safeLastUpdate(part, update) &&
-                isUserTurn;
+            return debug(part.listMoves != update.listMoves) &&
+                debug(isValidPartUpdateDuringGame(part)) &&
+                debug(safeTurnIncrease(part.turn, update.turn)) &&
+                debug(safeLastUpdate(part, update)) &&
+                setsWinnerIfVictory(part, update) &&
+                debug(isUserTurn);
+        }
+        function setsWinnerIfVictory(part, update) {
+            let isVictory = update.result == 3; // 3 = victory
+            let playerIsLoserOrWinner = update.loser.id == request.auth.uid || update.winner.id == request.auth.uid;
+            let otherPlayerIsLoserOrWinner = (update.winner.id == part.playerZero.id || update.winner.id == part.playerOne.id);
```

Comment:
si request.auth.uid == update.winner.id &&
    request.auth.uid == part.playerZero.id &&
    hence update.winer.id == part.playerZero.id
THEN who care who is playerOne, it still pass, no ?
p-e return playerZeroIsLoserOrWinner && playerOneIsLoserOrWinner et somme toute playerIsLoserOrWinner je crois que c'est testé upstream, c'est une des conditions les plus basiques du test ?

Reply:
updaté pour s'assurer qu'on ait bien et Zero et One dans les loser/player